### DISCUSSION TITLE
Option to ignore unresolved aliases

### DISCUSSION THREAD

**Discussion by rarkins** [eemeli/yaml#583] [Ideas]
We have a use case in Renovate Bot where users would like Renovate to ignore unresolved aliases and continue anyway: https://github.com/renovatebot/renovate/discussions/31902

Is there any technical or philosophical reason why this option couldn't be supported?

**Comment by eemeli** [comment]
If you need special handling for unresolved aliases, why not define a visitor to apply it? For an example, see https://github.com/eemeli/yaml/issues/497#issuecomment-2408918001.

**Reply by rarkins** [reply]
Thanks for the quick response. Do you mean something like this?

```js
import { parseDocument, visit } from 'yaml'

const doc = parseDocument('foo: *bar');
visit(doc, {
  Alias(key, node) {
    if (!node.resolve(doc)) return undefined
  }
});
```

Would you expect that to fix/remove any `doc.errors`? I attempted it with this code but it didn't work:

```js
  const rawDocuments = parseAllDocuments(
    massagedContent,
    prepareParseOption(options),
  );

  const schema = options?.customSchema;

  const results: ResT[] = [];
  for (const rawDocument of rawDocuments) {
    visit(rawDocument, {
      Alias(key, node) {
        if (!node.resolve(rawDocument)) {
          return undefined;
        }
      },
    });

    const document = rawDocument.toJS({ maxAliasCount: 10000 });

    const errors = rawDocument.errors;
    // handle YAML parse errors
    if (errors?.length) {
      const error = new AggregateError(errors, 'Failed to parse YAML file');
      if (options?.failureBehaviour === 'filter') {
        logger.debug({ error, document }, 'Failed to parse YAML');
        continue;
      }
      throw error;
    }
```

**Reply by eemeli** [reply]
I meant applying some change to the node or document in the visitor. As is, returning `undefined` does nothing. If you want to replace the node, create its replacement and return that. See the docs here: https://eemeli.org/yaml/#finding-and-modifying-nodes

**Reply by rarkins** [reply]
If done correctly, will modifying nodes automatically resolve associated errors?

**Reply by eemeli** [reply]
Within this library, yes, because alias resolution is only done when calling `toJS()`, but in your larger application context I've no idea what errors you might end up experiencing because of these unresolved aliases.
